aries-site/src/examples/templates/forms/managing-child-objects/CreateCluster.js

```diff
@@ -0,0 +1,243 @@
+
+const hostTemplate = {
+  name: '',
+  host: '',
+  cpu: '',
+  memory: '',
+};
+
+const INPUT_MAP = {
+  name: ({ key, index, ...rest }) => (
+    <FormField
+      key={key}
+      htmlFor={`hosts[${index}].name`}
+      name={`hosts[${index}].name`}
+      label="Host name"
+    >
+      <TextInput
+        id={`hosts[${index}].name`}
+        name={`hosts[${index}].name`}
+        {...rest}
+      />
+    </FormField>
+  ),
+  host: ({ key, index, ...rest }) => (
+    <FormField
+      key={key}
+      htmlFor={`hosts[${index}].host`}
+      name={`hosts[${index}].host`}
+      label="Host address"
+      required
+      aria-required="true"
+    >
+      <TextInput
+        id={`hosts[${index}].host`}
+        name={`hosts[${index}].host`}
+        {...rest}
+      />
+    </FormField>
+  ),
+  cpu: ({ key, index, ...rest }) => (
+    <Box key={key} width="xsmall">
+      <FormField
+        htmlFor={`hosts[${index}].cpu`}
+        name={`hosts[${index}].cpu`}
+        label="CPU cores"
+        required
+        aria-required="true"
+      >
+        {/* Make this a masked input */}
+        <Select
+          id={`hosts[${index}].cpu`}
+          name={`hosts[${index}].cpu`}
+          options={[2, 4, 6, 8]}
+          {...rest}
+        />
+      </FormField>
+    </Box>
+  ),
+  memory: ({ key, index, ...rest }) => (
+    <Box key={key} width="xsmall">
+      <FormField
+        key={key}
+        htmlFor={`hosts[${index}].memory`}
+        name={`hosts[${index}].memory`}
+        label="Memory (GB)"
+        required
+        aria-required="true"
+      >
+        <Select
+          id={`hosts[${index}].memory`}
+          name={`hosts[${index}].memory`}
+          options={[32, 64, 128, 256, 512]}
+          {...rest}
+        />
+      </FormField>
+    </Box>
+  ),
+};
+
+const defaultValues = {
+  'cluster-name': '',
+  'resource-manager': false,
+  'automation-level': '',
+  'migration-threshold': 15,
+  hosts: [{ ...hostTemplate }],
+};
+
+export const CreateCluster = () => {
+  const [formValues, setFormValues] = useState(defaultValues);
+
+  const handleAdd = () => {
+    const nextHosts = [...formValues.hosts, { ...hostTemplate }];
+    setFormValues({ ...formValues, hosts: nextHosts });
+  };
+
+  const handleRemove = index => {
+    if (formValues.hosts?.length > 0) {
+      setFormValues({
+        ...formValues,
+        hosts: formValues.hosts.filter((value, i) => i !== index),
+      });
+    }
+  };
+
+  const handleRemoveAll = () => {
+    setFormValues({
+      ...formValues,
+      hosts: [],
+    });
+  };
+
+  const onChange = value => {
+    setFormValues(value);
+  };
+
+  const onSubmit = event => {
+    console.log(event.value);
+  };
+
+  return (
+    <Box alignSelf="center" gap="medium" width="medium">
+      <Heading level={2} margin="none">
+        Create Cluster
+      </Heading>
+      <Form
+        value={formValues}
+        onChange={onChange}
+        onSubmit={onSubmit}
+        validate="blur"
```

Comment:
I noticed were doing `validate="blur"` is this what we are recommending for this type of pattern?

Reply:
I will need to confirm that with Jessica and Kenny.

Reply:
[This ticket](https://github.com/grommet/hpe-design-system/issues/2899) mentioned using validate on blur within the success criteria so I just stuck with that when creating the prototype in codesnadbox

Reply:
Thanks for clarifying. I'll leave the `blur` validation pattern.

I did notice there's a Grommet bug with `blur` validation when you use Drop related components like Select. When you open the Select, it registers that as a "blur" event and causes the error state because no value has been selected. I think if the focus is blurring from the formfield to the drop portal, it shouldn't flag the validation. Related to this existing issue: https://github.com/grommet/grommet/issues/5538 and proposing we don't let that Grommet fix be a blocker on this PR.

Reply:
yeah I also noticed that 👆  (kinda why I asked) lol
I just wasn't sure since it was on every example if we wanted to recommend that but I think that behavior is also up to product teams..

Reply:
In general, we want to provide users feedback as soon as we have information to provide that feedback -- that way they can correct errors in their flow and not have to go back to fix errors later on. In a lengthy context like this, having to go back a fix an error would be especially annoying. That said, `onChange` would be the quickest feedback, however when used against text input where the user hasn't even had a chance to finish typing onChange presents errors prematurely. A debounce solution + onChange might be the best way to go -- `onBlur` a close second.

Thanks for tracking down the onBlur bug.